io/src/main/scala/sbt/nio/file/PathFilter.scala

```diff
@@ -0,0 +1,259 @@
+/*
+ * sbt IO
+ *
+ * Copyright 2011 - 2019, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ */
+
+package sbt.nio.file
+
+import java.io.FileFilter
+import java.nio.file.{ Files, Path }
+
+/**
+ * A filter for a path and its attributes.
+ */
+trait PathFilter {
+
+  /**
+   *
+   * @param path the path name
+   * @param attributes the file attributes corresponding to `path`
+   * @return true if the filter accepts the path
+   */
+  def accept(path: Path, attributes: FileAttributes): Boolean
+}
+
+object PathFilter {
+
+  /**
+   * Returns a PathFilter that accepts any path for which there exists a [[Glob]] that matches
+   * the path
+   * @param globs the glob patterns
+   * @return a PathFilter that accepts any path matching one or more input [[Glob]]s.
+   */
+  def apply(globs: Glob*): PathFilter = if (globs.isEmpty) NoPass else new GlobPathFilter(globs)
+
+  /**
+   * Provides extension methods for [[PathFilter]]
+   * @param pathFilter the [[PathFilter]] to extend
+   */
+  implicit class Ops(val pathFilter: PathFilter) extends AnyVal {
+
+    /**
+     * Returns a combined [[PathFilter]] that returns true only if both [[PathFilter]] instances
+     * accept the path.
+     * @param other the other [[PathFilter]]
+     */
+    def &&(other: PathFilter): PathFilter = pathFilter match {
+      case NoPass  => NoPass
+      case AllPass => other
+      case f =>
+        other match {
+          case NoPass  => NoPass
+          case AllPass => f
+          case g       => new AndPathFilter(f, g)
+        }
+    }
+
+    /**
+     * Returns a combined [[PathFilter]] that returns true only if both [[PathFilter]] instances
+     * accept the path.
+     * @param other the other [[PathFilter]]
+     */
+    def ||(other: PathFilter): PathFilter = pathFilter match {
+      case NoPass  => other
+      case AllPass => AllPass
+      case f =>
+        other match {
+          case NoPass  => f
+          case AllPass => AllPass
+          case g       => new OrPathFilter(f, g)
+        }
+    }
+
+    /**
+     * Inverts the [[PathFilter.accept]] method.
+     * @return
+     */
+    def unary_! : PathFilter = pathFilter match {
+      case AllPass         => NoPass
+      case NoPass          => AllPass
+      case HiddenFilter    => NotHiddenFilter
+      case NotHiddenFilter => HiddenFilter
+      case pf              => new NotPathFilter(pf)
+    }
+  }
+
+  /**
+   * Converts an instance of [[sbt.io.FileFilter]] to an [[sbt.nio.file.PathFilter]]. It will
+   * de-structure the [[sbt.io.FileFilter]] if possible to convert it to an equivalent, and
+   * possibly more efficient, [[PathFilter]].
+   *
+   * @param fileFilter the filter to convert
+   * @return the converted.
+   */
+  implicit def fromFileFilter(fileFilter: FileFilter): PathFilter = fileFilter match {
+    case sbt.io.HiddenFileFilter    => HiddenFilter
```

Comment:
Would it make sense to keep the names same, since `DirectoryFilter` is `DirectoryFilter`. It gets confusing when are doing the same thing but named slightly differently like `HiddenFileFilter` vs `HiddenFilter`, `NothingFilter` vs `NoPass `